### PR TITLE
[MAINTENANCE] Remove BatchRequest from Rule-Based Profiler Configuration and from Builder Constructor Arguments

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
* Phase 1: `Batch` data in the form of `batch_request` or `batch_list` is supplied directly to the `run()` method, and not used as part of Profiler component (Builders) configuration nor in the list of Builder constructor arguments. 
* All usage and tests have been updated.

### Next -- Phase 2:
Eliminate dependence on `DataContext` in `DataAssistant` and in as many `Rule-Based Profiler` components as possible.

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- JIRA: GREAT-735/GREAT-816
-
-

After submitting your PR, CI checks will run and @ge-cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: tests/integration/profiling/rule_based_profilers/test_profiler_basic_workflows.py [hunk 1]
```diff
@@ -82,10 +82,9 @@ def test_domain_builder(data_context_with_taxi_data):
     )
     domain_builder: DomainBuilder = ColumnDomainBuilder(
         include_column_name_suffixes=["_amount"],
-        batch_request=batch_request,
         data_context=context,
     )
-    domains: list = domain_builder.get_domains()
+    domains: list = domain_builder.get_domains(batch_request=batch_request)
```
Comment (Shinnnyshinshin) [inline tests/integration/profiling/rule_based_profilers/test_profiler_basic_workflows.py:87]: :heart:

File: tests/integration/profiling/rule_based_profilers/test_profiler_user_workflows.py [hunk 2]
```diff
@@ -134,7 +134,13 @@ def test_alice_profiler_user_workflow_single_batch(
...
     )
 
-    profiler.run()
+    # BatchRequest yielding exactly one batch
+    alice_single_batch_data_batch_request: dict = {
+        "datasource_name": "alice_columnar_table_single_batch_datasource",
+        "data_connector_name": "alice_columnar_table_single_batch_data_connector",
+        "data_asset_name": "alice_columnar_table_single_batch_data_asset",
+    }
+    profiler.run(batch_request=alice_single_batch_data_batch_request)
```
Comment (Shinnnyshinshin) [inline tests/integration/profiling/rule_based_profilers/test_profiler_user_workflows.py:143]: I must say, I really do like this flow a lot better. Thank you @alexsherstinsky

File: tests/rule_based_profiler/data_assistant/test_volume_data_assistant.py [hunk 3]
```diff
@@ -2240,15 +2240,13 @@ def test_get_metrics_and_expectations(
...
                                 "module_name": "great_expectations.rule_based_profiler.domain_builder.table_domain_builder",
-                                "batch_request": None,
                                 "class_name": "TableDomainBuilder",
                             },
                             "parameter_builders": [
                                 {
                                     "module_name": "great_expectations.rule_based_profiler.parameter_builder.metric_multi_batch_parameter_builder",
                                     "metric_name": "table.row_count",
                                     "json_serialize": True,
-                                    "batch_request": None,
```
Comment (Shinnnyshinshin) [inline tests/rule_based_profiler/data_assistant/test_volume_data_assistant.py:2251]: I really think this makes more sense. Thank you

File: tests/rule_based_profiler/parameter_builder/test_regex_pattern_string_parameter_builder.py [hunk 4]
```diff
@@ -487,6 +486,7 @@ def test_regex_two_candidates(mock_data_context: mock.MagicMock, batch_fixture:
     regex_pattern_string_parameter_builder.build_parameters(
         domain=domain,
         parameters=parameters,
+        batch_list=[batch],
```
Comment (Shinnnyshinshin) [inline tests/rule_based_profiler/parameter_builder/test_regex_pattern_string_parameter_builder.py:489]: :heart:

File: tests/test_fixtures/rule_based_profiler/quentin_user_workflow_verbose_profiler_config.yml [hunk 5]
```diff
@@ -13,33 +13,17 @@ variables:
   num_bootstrap_samples: 9139
   bootstrap_random_seed: 43792
 
-  # BatchRequest yielding thirty five (35) batches (January, 2018 -- November, 2020 trip data)
```
Comment (Shinnnyshinshin) [inline tests/test_fixtures/rule_based_profiler/quentin_user_workflow_verbose_profiler_config.yml:16]: Love that the configuration is being simplified, but would like to ensure the information in the in-line comment is preserved. What are your thoughts on including the in-line comment whenever this BatchRequest is used in other tests?

File: tests/test_fixtures/rule_based_profiler/quentin_user_workflow_verbose_profiler_config.yml [hunk 6]
```diff
@@ -13,33 +13,17 @@ variables:
...
-    data_connector_query:
-      index: ":-1"
-
   false_positive_rate: 5.0e-2
 
 rules:
   column_quantiles_rule:
     domain_builder:
       class_name: ColumnDomainBuilder
-      # BatchRequest yielding exactly one batch (November, 2020 trip data)
```
Comment (Shinnnyshinshin) [inline tests/test_fixtures/rule_based_profiler/quentin_user_workflow_verbose_profiler_config.yml:30]: Same thought here regarding preserving the knowledge.

**Review by Shinnnyshinshin — COMMENTED** [review]
Thank you very much for this work Alex. LGTM so far, and I'm excited to see what part 2 is going to bring.

Currently it looks like a few tests are failing still, and I left a comment regarding the information around batchrequests that would be wonderful to preserve. Please have a look and let me know what you think.

File: tests/integration/profiling/rule_based_profilers/test_profiler_user_workflows.py (hunk 2 above)
Comment (alexsherstinsky) [reply tests/integration/profiling/rule_based_profilers/test_profiler_user_workflows.py:143]: 🙇

File: tests/test_fixtures/rule_based_profiler/quentin_user_workflow_verbose_profiler_config.yml (hunk 5 above)
Comment (alexsherstinsky) [reply tests/test_fixtures/rule_based_profiler/quentin_user_workflow_verbose_profiler_config.yml:16]: @Shinnnyshinshin I recall putting this comment next to python version of these batch requests in the tests.

File: tests/test_fixtures/rule_based_profiler/quentin_user_workflow_verbose_profiler_config.yml (hunk 6 above)
Comment (alexsherstinsky) [reply tests/test_fixtures/rule_based_profiler/quentin_user_workflow_verbose_profiler_config.yml:30]: @Shinnnyshinshin I did just that (hopefully in sufficient number of places).

**Review by cdkini — APPROVED** [review]
LGTM

**Pull request merged**